src/app/components/core/dashboard/dashboard.component.ts

```diff
@@ -1,106 +1,99 @@
-  public invalidLimitSize = false;
+  private timerSubscription: AnonymousSubscription;
+  private postsSubscription: AnonymousSubscription;
+
+  public refreshTimer = GRAPH_REFRESH_INTERVAL;
 
-  constructor(private statisticsService: StatisticsService, private alertService: AlertService, public ngProgress: NgProgress) { }
+  DEFAULT_LIMIT = 20;
+
+  constructor(private statisticsService: StatisticsService, private alertService: AlertService, private ping: PingService) { }
 
   ngOnInit() {
+    // To check if data saved in valid format in local storage
+    const optedGraphStorage = JSON.parse(localStorage.getItem('OPTED_GRAPHS'));
+    console.log(typeof (optedGraphStorage), optedGraphStorage);
+    if (optedGraphStorage.length === 0  || typeof (optedGraphStorage[0]) !== 'object') {
+      localStorage.removeItem('OPTED_GRAPHS');
+    }
     this.getStatistics();
+    this.ping.refreshIntervalChanged.subscribe((timeInterval: number) => {
+      this.refreshTimer = timeInterval;
+    });
   }
 
-  public showGraph(graphs) {
-    this.selectedKeys = [];
-    // get keys selected from dropdown
-    for (const k of graphs) {
-      this.selectedKeys.push(k.itemName);
+  public showGraph(selectedGraph) {
+    // get keys selected from drop down
+    this.statisticsKeys.map((item) => (item.key === selectedGraph.key && selectedGraph.checked === false) ? item.checked = false : true);
+    if (selectedGraph.checked === false && this.selectedKeys.length > 0) {
+      this.selectedKeys = this.selectedKeys.filter((dt => dt.key !== selectedGraph.key));
+    } else {
+      this.selectedKeys.push(selectedGraph);
     }
-    // save keys in local storage
-    localStorage.setItem('OPTED_GRAPHS', JSON.stringify(this.selectedKeys));
 
+    // if there is no graph selected the set default to READINGS and PURGED
+    if (this.selectedKeys.length === 0) {
+      this.selectedKeys = [
+        { key: 'READINGS', checked: true },
+        { key: 'PURGED', checked: true },
+        { key: 'North Readings to PI', checked: true }];
+    }
+
+    localStorage.setItem('OPTED_GRAPHS', JSON.stringify(this.selectedKeys));
     this.graphsToShow = [];
-    for (const k of this.selectedKeys) {
+    for (const dt of this.selectedKeys) {
       const selectedKeyData = [];
-      selectedKeyData.push(this.statistics.filter(value => value['itemName'] === k));
+      selectedKeyData.push(this.statistics.filter(value => value['itemName'] === dt.key));
       this.graphsToShow.push(selectedKeyData[0][0]);
     }
     this.getStatistics();
   }
 
-  getLimitBasedGraph(limit, key) {
-    this.invalidLimitSize = false;
-    if (limit === null || limit === undefined) {
-      limit = this.DEFAULT_LIMIT;
-    }
-
-    if (limit > this.MAX_RANGE) {
-      this.invalidLimitSize = true; // limit range validation
-      return;
-    }
-    this.refreshGraph(limit, key);
-  }
-
   public getStatistics(): void {
-    /** request started */
-    this.ngProgress.start();
```

Comment:
why this has been removed?

Reply:
Because graph auto refresh at every 5 second interval and progress bar look annoying  on every  call to refresh graph.